akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala

```diff
@@ -858,12 +858,10 @@ final class Partition[T](val outputPorts: Int, val partitioner: T => Int, val ea
                   if (idx == outPendingIdx) {
                     push(o, elem)
                     outPendingElem = null
-                    if (!isClosed(in)) {
-                      if (!hasBeenPulled(in)) {
-                        pull(in)
-                      }
-                    } else
+                    if (isClosed(in))
                       completeStage()
+                    else if (!hasBeenPulled(in))
+                      pull(in)
                   }
                 } else if (!hasBeenPulled(in))
```

Comment:
I was thinking and trying to come up with a test where this would also be wrong. That `in` is closed here, but couldn't. It's probably fine.

Reply:
Hmmm, this path is a pull on this partition-downstream when there is a pending element but that element is for another partition-downstream. Should it actually even pull in that scenario, when we already have one buffered element for a downstream which isn't ready to accept it?

Won't that cause the buffered element to be lost because of the assignment on L830?

Reply:
it will never do this when there is a pending, L856

Reply:
Ah, thx, I missed that.